test/quic_stream_test.c

```diff
@@ -321,9 +321,161 @@ static int test_bulk(int idx)
+                                                   10, 1))
+        || !TEST_true(ossl_quic_rstream_queue_data(rstream, NULL, 0,
+                                                   simple_data, 1, 0))
+        || !TEST_true(ossl_quic_rstream_peek(rstream, buf, sizeof(buf),
+                                             &readbytes, &fin))
+        || !TEST_false(fin)
+        || !TEST_size_t_eq(readbytes, 1)
+        || !TEST_mem_eq(buf, 1, simple_data, 1)
+        || !TEST_true(ossl_quic_rstream_queue_data(rstream, NULL,
+                                                   0, simple_data,
+                                                   10, 0))
+        || !TEST_true(ossl_quic_rstream_peek(rstream, buf, sizeof(buf),
+                                             &readbytes, &fin))
+        || !TEST_false(fin)
+        || !TEST_size_t_eq(readbytes, 15)
+        || !TEST_mem_eq(buf, 15, simple_data, 15)
+        || !TEST_true(ossl_quic_rstream_queue_data(rstream, NULL,
+                                                   15,
+                                                   simple_data + 15,
+                                                   sizeof(simple_data) - 15, 1))
+        || !TEST_true(ossl_quic_rstream_available(rstream, &avail, &fin))
+        || !TEST_true(fin)
+        || !TEST_size_t_eq(avail, sizeof(simple_data))
+        || !TEST_true(ossl_quic_rstream_read(rstream, buf, 12,
+                                             &readbytes, &fin))
+        || !TEST_false(fin)
+        || !TEST_size_t_eq(readbytes, 12)
+        || !TEST_mem_eq(buf, 12, simple_data, 12)
+        || !TEST_true(ossl_quic_rstream_read(rstream, buf + 12, sizeof(buf) - 12,
+                                             &readbytes, &fin))
+        || !TEST_true(fin)
+        || !TEST_size_t_eq(readbytes, sizeof(buf) - 12)
+        || !TEST_mem_eq(buf, sizeof(buf), simple_data, sizeof(simple_data))
+        || !TEST_true(ossl_quic_rstream_read(rstream, buf, sizeof(buf),
+                                             &readbytes, &fin))
+        || !TEST_true(fin)
+        || !TEST_size_t_eq(readbytes, 0))
+        goto err;
+
+    ret = 1;
+
+ err:
+    ossl_quic_rstream_free(rstream);
+    return ret;
+}
+
+static int test_rstream_random(int idx)
+{
+    unsigned char *bulk_data = NULL;
+    unsigned char *read_buf = NULL;
+    QUIC_RSTREAM *rstream = NULL;
+    size_t i, read_off, queued_min;
+    const size_t data_size = 10000;
+    int r, s, fin;
+    int ret = 0;
+    size_t readbytes = 0;
+
+    if (!TEST_ptr(bulk_data = OPENSSL_malloc(data_size))
+        || !TEST_ptr(read_buf = OPENSSL_malloc(data_size))
+        || !TEST_ptr(rstream = ossl_quic_rstream_new(NULL, NULL, NULL)))
+        goto err;
+
+    for (i = 0; i < data_size; ++i)
+        bulk_data[i] = (unsigned char)(test_random() & 0xFF);
+
+    i = read_off = queued_min = 0;
+    for (r = 0; r < 100; ++r) {
+        for (s = 0; s < 10; ++s) {
+            size_t off = i * 10, size = 10;
```

Comment:
`i` never seems to get incremented in this test

Reply:
Fixed, this was actually unfinished.